portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -56,19 +57,30 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int numCommentDisplay = getNumDisplay(request);
+    if (numCommentDisplay == -1) {
+      response.setContentType("text/html");
+      response.getWriter().println("Please enter an integer between 1 and 10.");
+      return;
+    }
+
     Query query = new Query(COMMENT_PARAM).addSort(TIMESTAMP_PROP, SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
     List<Comment> comments = new ArrayList<>();
+    int i = 0;
     for (Entity entity : results.asIterable()) {
-      long id = entity.getKey().getId();
-      String commentText = (String) entity.getProperty(TEXT_PROP);
-      long timestamp = (long) entity.getProperty(TIMESTAMP_PROP);
+      if (i < numCommentDisplay) {
```

Comment:
With this logic, you'll loop through every `entity` in `results` even if `results` is really long in comparison to `numCommentDisplay`. Instead, I'd suggest doing something similar to:
```
if (i >= numCommentDisplay) {
  break;
}
```

Reply:
To add on to this, `i` is just the length of the `List`, so I'd replace this with

```java
if (comments.size() >= numCommentDisplay) {
  break;
}
```
and get rid of `i` entirely.